File: archive/scripts/validate_pack_a.py
import json
import re
from pathlib import Path

file_path = Path('pack_a_corrected.js')
content = file_path.read_text(encoding='utf-8')
print('File size:', len(content))
print('Lines:', len(content.splitlines()))

# Count question objects
question_ids = re.findall(r'"QuestionID"\s*:\s*"([^"]+)"', content)
print('Question IDs found:', len(question_ids))
print('Unique IDs:', len(set(question_ids)))
print('First 5 IDs:', question_ids[:5])

# Check for difficulty values
difficulties = re.findall(r'"Difficulty"\s*:\s*"([^"]+)"', content)
print('Difficulty values:', set(difficulties))

# Check for item types
item_types = re.findall(r'"ItemType"\s*:\s*"([^"]+)"', content)
print('Item types:', set(item_types))

# Check for sections
sections = re.findall(r'"Section"\s*:\s*"([^"]+)"', content)
print('Sections:', set(sections))

File: archive/scripts/validate_pack_a.js
const fs = require('fs');

const content = fs.readFileSync('pack_a_corrected.js', 'utf8');
console.log('File size:', content.length);
console.log('Lines:', content.split('\n').length);

// Count question objects
const questionIds = content.match(/"QuestionID"\s*:\s*"([^"]+)"/g);
const ids = questionIds ? questionIds.map(m => m.match(/"QuestionID"\s*:\s*"([^"]+)"/)[1]) : [];
console.log('Question IDs found:', ids.length);
console.log('Unique IDs:', new Set(ids).size);
console.log('First 5 IDs:', ids.slice(0, 5));

// Check for difficulty values
const difficulties = content.match(/"Difficulty"\s*:\s*"([^"]+)"/g);
const diffValues = difficulties ? difficulties.map(m => m.match(/"Difficulty"\s*:\s*"([^"]+)"/)[1]) : [];
console.log('Difficulty values:', [...new Set(diffValues)]);

// Check for item types
const itemTypes = content.match(/"ItemType"\s*:\s*"([^"]+)"/g);
const itemTypeValues = itemTypes ? itemTypes.map(m => m.match(/"ItemType"\s*:\s*"([^"]+)"/)[1]) : [];
console.log('Item types:', [...new Set(itemTypeValues)]);

// Check for sections
const sections = content.match(/"Section"\s*:\s*"([^"]+)"/g);
const sectionValues = sections ? sections.map(m => m.match(/"Section"\s*:\s*"([^"]+)"/)[1]) : [];
console.log('Sections:', [...new Set(sectionValues)]);